out_writeapi.go

```diff
@@ -567,16 +580,35 @@ func FLBPluginFlushCtx(ctx, data unsafe.Pointer, length C.int, tag *C.char) int
 		return output.FLB_ERROR
 	}
 	// Holds stream slice for ease of use
-	streamSlice := *config.managedStreamSlice
+	streamSlice := config.managedStreamSlice
 
 	// checks responses for each stream using a loop
-	sliceLen := len(streamSlice)
+	sliceLen := len(*streamSlice)
 	for i := 0; i < sliceLen; i++ {
-		// TODO: Use return from check resopnses to build stream if necessary
-		checkResponses(ms_ctx, (streamSlice)[i].appendResults, false, &config.mutex, config.exactlyOnce, id)
+		checkResponses(ms_ctx, (*streamSlice)[i].appendResults, false, &config.mutex, config.exactlyOnce, id)
 	}
 
-	// TODO: Build a NewManagedStream if request count is above threshold
+	// Gets stream with least values in queue
+	mostEfficient := getStreamIndex(config.managedStreamSlice)
```

Comment:
Reading the slice needs mutex protection too (not just appending to it). 
In other places too.